README.md

```diff
@@ -1,32 +1,158 @@
-# python-flask-template
+OpenFaaS Python Flask Templates
+=============================================
 
-Python OpenFaaS template with Flask
+The Python Flask templates make use of the incubator project [of-watchdog](https://github.com/openfaas-incubator/of-watchdog) to handle higher throughput than the [classic watchdog](https://github.com/openfaas/faas/tree/master/watchdog) due to the process being kept warm.
 
-To try this out with either Python 2.7 or Python 3.6:
-
-```bash
-faas template pull https://github.com/openfaas-incubator/python-flask-template
-faas new --list
-Languages available as templates:
+Templates available under this repository:
 - python27-flask
 - python3-flask
-```
+- python3-flask-armhf
+- python3-http*
+- python3-http-armhf*
 
-Generate a function with one of the languages:
+Notes: 
+- The templates listed with an asterik are the only ones that support additional control over the HTTP response and provide access to HTTP request details.
+- To build and deploy a function for Raspberry Pi or ARMv7 in general, use the language templates ending in *-armhf*
 
-```bash
-faas new --lang python3-flask myfunction
-mv myfunction.yml stack.yml
+## Downloading the templates
+```
+$ faas template pull https://github.com/openfaas-incubator/python-flask-template
 ```
 
-Followed by the usual flow:
+# Using the python27-flask/python3-flask templates
+Create a new function
+```
+$ faas new --lang python27-flask <fn-name>
+```
+Build, push, and deploy
+```
+$ faas up -f <fn-name>.yml
+```
+Test the new function
+```
+$ echo -n content | faas invoke <fn-name>
+```
 
+# Using the python3-http templates
+Create a new function
+```
+$ faas new --lang python3-http <fn-name>
+```
+Build, push, and deploy
+```
+$ faas up -f <fn-name>.yml
+```
+Set your OpenFaaS gateway URL. For example:
 ```
-faas build \
-  && faas deploy
-  && faas list --verbose
+$ OPENFAAS_URL=http://127.0.0.1:8080
+```
+Test the new function
+```
+$ curl -i $OPENFAAS_URL/function/<fn-name>
+```
+
+## Event and Context Data
```

Comment:
Is there an "official doc" where the contract with the watchdog is explained?
The goal is to know which field is mandatory, and which is not

Reply:
Hi @pcorbel how is this question related to the Python template?